code/drasil-lang/Language/Drasil/Expr.hs

```diff
@@ -136,10 +136,8 @@ data Expr where
   --   of an 'Expr'.  Could be called BigOp.
   --   ex: Summation is represented via 'Add' over a discrete domain.
   Operator :: AssocArithOper -> DomainDesc Expr Expr -> Expr -> Expr
-  -- | The expression is an element of a space.
-  -- IsIn     :: Expr -> Space -> Expr
-  -- | A different kind of 'IsIn'. A 'UID' is an element of an interval.
-  RealI    :: UID -> RealInterval Expr Expr -> Expr
+  -- | A different kind of 'IsIn'. An 'Expr' is an element of an interval.
+  RealI    :: Expr -> RealInterval Expr Expr -> Expr
```

Comment:
The reason for UID is that we really only want 'names' as the first component. It's almost ok if the data-structure allows more, as long as the smart constructor does not.

Reply:
I was thinking that this could add in extra flexibility to have those non-name cases, e.g., `1 < (x / 2pi) < 5`. However, I now also think there is also benefit to changing it to Expr for when we add typing. It seems a bit difficult to "add type information" from UIDs. Having Exprs instead, where the typing is will already be done by the `sy` (though, we might need to later replace `sy` with variants for each type or somehow allow it to gain the type information from UIDs), will allow for later processing to be done a bit smoother.

I started writing some general thoughts on a new [jasonWriting](https://github.com/JacquesCarette/Drasil/tree/jasonWriting) branch and coming across `C`, I was thinking that we would need to do something like this:
https://github.com/JacquesCarette/Drasil/blob/846c0308c87b675ccee16cca4c54f2da5de51e14/People/Jason/Typing.lhs#L83-L105
So I think this will, at least, allow us to not have to later split RealI's smart constructors into multiple ones as well to create the typing information as well.

The smart constructor currently does not allow the extra flexibility. Would there be any issues if we were to allow it?

Reply:
So I'm not sure we want this extra flexibility! I would want to rewrite such things to isolate the `x`. We want `RealI` to mean "bounding interval for a variable".

Reply:
Ah, ok, that makes sense. In that case, holding the UID would definitely be better than carrying it in an `C`. I think we can cancel this PR on this alone then.